ko/security.rst

```diff
@@ -18,8 +18,300 @@ CUBRID 보안
+
+.. code-block:: bash
+
+	$ cubrid tde --show-keys testdb
+	Key File: /home/usr/CUBRID/databases/testdb/testdb_keys
+
+	The current key set on testdb:
+	Key Index: 2
+	Created on Fri Nov 27 11:14:54 2020
+	Set     on Fri Nov 27 11:15:30 2020
+
+	Keys Information: 
+	Key Index: 0 created on Fri Nov 27 11:11:27 2020
+	Key Index: 1 created on Fri Nov 27 11:14:47 2020
+	Key Index: 2 created on Fri Nov 27 11:14:54 2020
+	Key Index: 3 created on Fri Nov 27 11:14:55 2020
+
+	The number of keys: 4
+
+.. note:: **기존의 키 파일을 사용하는 데이터베이스 생성**
+
+	키를 변경하기 위해서는  데이터베이스에 등록된 키를 가지고 있어야 한다. 그 때문에 다른 키 파일을 사용하도록 변경하는 것은 쉽지 않다. 만약 여러 데이터베이스가 하나의 키 파일을 공유하도록 하고 싶다거나, 기존에 생성되어 있던 키 파일을 사용하도록 하고 싶다면 **tde_keys_file_path** 시스템 파라미터를 통하여 해당 키 파일의 경로를 설정한 상태에서 데이터베이스를 생성하도록 한다. 해당 시스템 파라미터에 대한 자세한 내용은 :ref:`other-parameters` 를 참고한다.
+
+.. _tde-enc-target:
+
+암호화 대상
+-----------
+
+.. _tde-enc-perm:
+
+영구 데이터 암호화
+^^^^^^^^^^^^^^^^^^
+
+암호화된 테이블이 생성되면 해당 테이블의 모든 유저 데이터와 테이블에 생성된 모든 인덱스의 유저 데이터가 암호화된다. 암호화 여부는 상속되지 않으며, CREATE TABLE LIKE 구문 사용 시에도 복사되지 않는다.
+
+.. _tde-enc-temp:
+
+임시 데이터 암호화
+^^^^^^^^^^^^^^^^^^
+
+테이블과 같은 영구 데이터 외에도 질의 수행 중에 생성되는 임시 데이터 또한 디스크에 저장될 수 있다. 이렇게 생성된 임시 데이터가 만약 암호화된 테이블의 데이터를 포함하고 있다면 마찬가지로 디스크에는 암호화된 상태로 쓰인다. 예를 들어 `SELECT * FROM tde_tbl ORDER BY att1` 과 같은 질의를 수행하거나 `tde_tbl` 에 인덱스를 생성하는 과정 등에서 생성된 임시 데이터는 모두 암호화되어 디스크에 저장된다. 임시 데이터에 대한 자세한 내용은 :ref:`temporary-volumes` 를 참고한다.
+
+.. _tde-enc-log:
+
+로그 데이터 암호화
+^^^^^^^^^^^^^^^^^^
+
+암호화된 테이블을 변경하면서 생성되는 REDO, UNDO 로그 레코드는 테이블의 유저 데이터를 포함할 수 있다. 큐브리드는 이러한 암호화 테이블에 대한 모든 로그 정보들에 대한 암호화도 제공하며, 활성 로그와 보관 로그에 모두 적용된다. 로그 볼륨에 대한 자세한 내용은 :ref:`database-volume` 을 참고한다.
+
+.. _tde-enc-dwb:
+
+DWB 암호화
+^^^^^^^^^^
+
+영구 데이터는 데이터 볼륨에 쓰이기 전에 이중 쓰기 버퍼 (Double Write Buffer, DWB)에 일시적으로 쓰는데, 이때에도 암호화된 테이블에 대한 데이터를 포함하고 있다면 암호화된다. DWB에 대한 자세한 내용은 :ref:`database-volume` 을 참고한다.
+
+.. _tde-enc-backup:
+
+백업 볼륨 암호화
+^^^^^^^^^^^^^^^^
+
+백업 볼륨은 데이터 볼륨과 로그 볼륨을 포함할 수 있는데, 각각의 볼륨이 암호화된 데이터를 포함하고 있다면 마찬가지로 암호화된 상태로 백업 볼륨에 포함된다. 백업에 대한 자세한 내용은 :ref:`backupdb` 를 참고한다. 
+
+**백업 키 파일**
+
+백업 볼륨에는 기본적으로 키 파일이 포함된다. 키 파일을 포함한 백업 볼륨이 유출될 경우, 볼륨 내의 데이터들은 암호화되어 있지만, 마스터 키도 함께 유출되므로 보안상의 문제가 있을 수 있다. 이를 방지하기 위해서는 **\\-\\-separate-keys** 를 통해 키를 분리하여 백업할 수 있다. 하지만, 이렇게 키 파일을 분리할 경우에는 분리된 키 파일을 유실해서는 안 된다. 분리된 백업 키 파일은 백업 볼륨과 같은 경로에 생성되며 **[database_name]_bk[backup_level]_keys** 라는 이름을 가진다.
+
+.. code-block:: bash
+
+	$ cubrid backupdb -S --separate-keys testdb 
+	Backup Volume Label: Level: 0, Unit: 0, Database testdb, Backup Time: Mon Nov 30 14:34:49 2020
+	$ ls
+	lob  testdb  testdb_bk0_keys  testdb_bk0v000  testdb_bkvinf  testdb_keys
+	testdb_lgar_t  testdb_lgat  testdb_lginf  testdb_vinf
+
+**백업 복구 시 키 파일 선택**
+
+백업 시에 분리된 키 파일은 백업 복구 (restoredb)를 수행할 때 **\\-\\-keys-file-path** 를 통해 복구 시 사용할 키 파일로 지정해 줄 수 있다. 지정한 경로에 올바른 키 파일이 존재하지 않을 경우 백업 복구는 실패한다.
```

Comment:
I don't think your suggestion is a proper explanation because the key file given by this option is not **restored**, but used during restore. What do you think?

Reply:
OK, do what you think.